### PR TITLE
Update all links to HTTPS

### PR DESCRIPTION
**Pull request by Bisaloo**
This update was carried out manually and each single link was checked for active mixed content or HTTPS misconfiguration.

Only links that stayed with HTTP:

- `packages.ropensci.org`, see https://github.com/ropensci/drat/issues/9

https://github.com/ropensci/dev_guide/blob/8aca66ea22114cb77a68a7c53a3ededcdd2eca11/softwarereview_intro.Rmd#L23

- `marker.to`  because of mixed content blocking

https://github.com/ropensci/dev_guide/blob/8aca66ea22114cb77a68a7c53a3ededcdd2eca11/pkg_ci.Rmd#L45

- `r-pkgs.had.co.nz` and `adv-r.had.co.nz` because HTTPS is not configured but I think these links should be updated to https://r-pkgs.org/ and https://adv-r.hadley.nz/ respectively. I can do this in this PR or in another one. Whatever is more convenient for you.

### CODE REVIEW

**Comment by maelle** [comment]
Wow thank you so much!

> I can do this in this PR or in another one. Whatever is more convenient for you.

In this PR please.

Can you also add an item in the NEWS section of Appendix.Rmd? And make sure your name is listed on the last line of index.Rmd? Thanks!

**Comment by Bisaloo** [comment]
No problem :smile:! I may be slightly obsessive about using HTTPS so you're totally welcome.

File: appendix.Rmd [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 ## dev
 
+* 2019-05-22, update all links eligible to HTTPS and update links to the latest versions of Hadley's books ([`@bisaloo`](https://github.com/bisaloo/), #167)
```
Comment (maelle) [inline appendix.Rmd]: maybe write "Hadley Wickham"?

Comment (maelle) [reply appendix.Rmd]: oh and now the book is also by Jenny Bryan so I'd suggest writing both names :-)

**Comment by maelle** [comment]
Lucky us then! :smile_cat:

**Comment by maelle** [comment]
Many thanks!

**Pull request merged**